### ISSUE TITLE
[RRFC] `.npmignore` pragma to include `.gitignore`

### ISSUE THREAD

**Issue by ljharb** [npm/rfcs#363]
## Motivation ("The Why")
While opinions differ, I consider "files" to be dangerous, since I have zero concern for accidental publishing but I have infinite concern for accidental breakage. (Let's avoid debating this)

When using `.npmignore`, one has to remember to start by `cp .gitignore .npmignore`, and then later, to keep those in sync, since unfortunately npm ignores .gitignore when .npmignore is present. This is a maintenance burden and is error-prone over time.

### Example/How/Desired Behavior
These examples show a hypothetical where "coverage data" is ignored by both git and npm, but the "dist" folder (build output) is ignored by git but included in npm.
<details><summary>Current:</summary>

`.gitignore`
```sh
coverage/ # coverage data
dist/ # build output
```

`.npmignore`
```sh
coverage/ # coverage data
```
</details>

<details><summary>Potential:</summary>

`.gitignore`
```sh
coverage/ # coverage data
dist/ # build output
```

`.npmignore`
```sh
<include gitignore> # ideally something that would make the npm CLI crash prior to supporting this feature
!dist/ # unignore build output
```
</details>

The advantage would be that additions to `.gitignore` no longer need to be propagated to `.npmignore`. The disadvantage would be that removals from `.gitignore` now do need to be propagated, but the explicit opt-in would mean that's expected.

Thoughts?

**Comment by isaacs** [comment]
I think this is pretty useful.  The challenge will be to find a syntax that would indeed make the npm CLI crash prior to supporting this feature.  Globs are _very_ accommodating, unfortunately, since filenames on posix systems can contain almost any character.  I don't see anything offhand that will make `ignore-walk` throw an error.  Anything that doesn't parse as a glob is just treated as part of the filename being matched.

**Comment by ljharb** [comment]
Another alternative would be to make a new filename, that when present causes npmignore to be ignored - but i have no brilliant suggestions for a name for it :-/

**Comment by ljharb** [comment]
Yeah, fair point - for npm back compat i'd still need to maintain an npmignore in that case, so that's probably not a viable solution.
